src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -91,10 +98,20 @@
               {:details
                (assoc (:details database) :version (:version db-metadata))}))
 
-(defn- cruft-dependent-columns [table-name]
-  ;; if this is a crufty table, mark initial sync as complete since we'll skip the subsequent sync steps
-  (let [is-crufty? (is-crufty-table? table-name)]
-    {:initial_sync_status (if is-crufty? "complete" "incomplete")
+(mu/defn- cruft-dependent-cols [{table-name :name :as table}
+                                database
+                                sync-stage :- [:enum ::reactivate ::create ::update]]
+  ;; if this is a crufty table, mark initial sync as complete since we'll be skipping the subsequent sync steps
+  (let [is-crufty? (crufty/name? table-name {:patterns        crufty-table-patterns
+                                             :pattern-strings (some-> database :settings :auto-cruft-tables)})]
```

Comment:
It might be slightly cleaner for this API to just take a set regexes? And combine them all into one set before passing them in?

Reply:
good idea, done